nodes/src/nodes/library/ssrf_protection.py

```
@@ -0,0 +1,282 @@
+
+def _build_allowlist(
+    extra: Optional[Sequence[str]] = None,
+) -> list[ipaddress.IPv4Network | ipaddress.IPv6Network]:
+    """Merge per-call allowlist with the global env-var allowlist."""
+    nets: list[ipaddress.IPv4Network | ipaddress.IPv6Network] = []
+
+    # Global allowlist from environment
+    env_val = os.environ.get(SSRF_ALLOWLIST_ENV, '').strip()
+    if env_val:
+        for cidr in env_val.split(','):
+            cidr = cidr.strip()
```

Ruff (0.15.7)

[warning] `for` loop variable `cidr` overwritten by assignment target

(PLW2901)